src/com/theironyard/novauc/Main.java

```diff
@@ -1,8 +1,256 @@
+                    int calories = Integer.valueOf(request.queryParams("calories"));
+                    int fat = Integer.valueOf(request.queryParams("fat"));
+                    int carbs = Integer.valueOf(request.queryParams("carbs"));
+                    int protein = Integer.valueOf(request.queryParams("protein"));
+                    String typeOfMeal = request.queryParams("typeOfMeal");
+                    int postId = currentUser.foodList.size();
+
+                    Food foodEntry = new Food(mealName,calories,carbs,fat,protein,typeOfMeal,userName,postId);
+
+                    currentUser.foodList.add(foodEntry);
+                    response.redirect("/");
+                    return "";
+                })
+                );
+
+        Spark.post("/delete-entry",
+                ((request, response) -> {
+                    Session session=request.session();
+                    String userName = session.attribute("userName");
+                    User currentUser = users.get(userName);
+                    if(currentUser == null){
+                        throw new Exception("User is not logged in");
+                    }
+                    if(request.queryParams("postdelete").equalsIgnoreCase("delete")){
+                        int deleteId = Integer.valueOf(request.queryParams("postId"));
+                        currentUser.foodList.remove(deleteId);
+                    }
+                    response.redirect("/");
+                    return "";
+                })
+                );
+
+        Spark.post("/edit-entry",
+                ((request, response) -> {
+                    Session session=request.session();
+                    String userName = session.attribute("userName");
+                    User currentUser = users.get(userName);
+                    if(currentUser == null){
+                        throw new Exception("User is not logged in");
+                    }
+                    String deleteFood = request.queryParams("deleteName");
+
+                    for(Food looper: currentUser.foodList){
+                        if(looper.getMealName().equalsIgnoreCase(deleteFood)){
+                            currentUser.foodList.remove(looper.getPostId());
+                        }
+                    }
+                    String mealName = request.queryParams("mealName");
+                    int calories = Integer.valueOf(request.queryParams("calories"));
+                    int fat = Integer.valueOf(request.queryParams("fat"));
+                    int carbs = Integer.valueOf(request.queryParams("carbs"));
+                    int protein = Integer.valueOf(request.queryParams("protein"));
+                    String typeOfMeal = request.queryParams("typeOfMeal");
+                    int postId = currentUser.foodList.size();
+
+                    Food foodEntry = new Food(mealName,calories,carbs,fat,protein,typeOfMeal,userName,postId);
```

Comment:
This edit logic is messy.  There are a couple of things that could go wrong here.  If they type in the wrong name it will add to the list and not remove what you were looking for.  A better way would be to assign the Food object you identify in your if statement on line 134 and use setter methods to change the different variables.